.ci/checker-framework.groovy

```diff
@@ -127,7 +128,10 @@ private static List<List<String>> getCheckerFrameworkErrors(final String profile
             checkerFrameworkLines.add(lineFromReader)
             lineFromReader = reader.readLine()
         }
-        process.waitFor()
+        int exitCode = process.waitFor()
+        if (exitCode != 0) {
+            throw new RuntimeException("Maven process exited with error code: " + exitCode)
+        }
```

Comment:
Not sure if we can really use this approach, since the build will fail when we have checker compilation errors too, and if we exit early we will not see the diff, right?

Reply:
A good approach my to be to figure out if the exit codes for OOMs/other exceptions is different from normal checker compilation failure. A good start to figuring this out might be to log the exit code here and keep track of the results as you go.

Reply:
yes, I see it now, not good but we can not do anything.
lets all back to catching of error in log.

Reply:
Got it, thanks.

Reply:
how do you see out of memmory error at https://github.com/checkstyle/checkstyle/actions/runs/7814021233/job/21314388563?pr=14409, I can not spot it.

Reply:
@relentless-pursuit it might be worth researching if we can set a custom exit code upon failure for the maven compiler plugin.

Reply:
Update: it looks like the build always returns success?

See: https://github.com/checkstyle/checkstyle/actions/runs/7593228559/job/20683478761, linked builds from the issue itself.

Even when we fail with OOM, build is "successful", so I assume that we return 0 whether we OOM, have normal checker failures, or not. We should explore configuring the maven compiler plugin to fail the build.

Reply:
> Update: it looks like the build always returns success?
> 
> See: https://github.com/checkstyle/checkstyle/actions/runs/7593228559/job/20683478761, linked builds from the issue itself.
> 
> Even when we fail with OOM, build is "successful", so I assume that we return 0 whether we OOM, have normal checker failures, or not. We should explore configuring the maven compiler plugin to fail the build.
> 
> Edit: ok, this is how we have the plugin set up:
> 
> ```
>               <failOnError>false</failOnError>
> ```
> 
> We could try to do:
> 
> ```
>               <failOnError>true</failOnError>
>               <failOnWarning>false</failOnWarning>
> ```
> 
> And see what happens.

Sure.